lib/internal/process/pre_execution.js

```diff
@@ -438,6 +440,47 @@ function setupInspectorHooks() {
   }
 }
 
+function setupNetworkInspection() {
+  if (!internalBinding('inspector').isEnabled() || !getOptionValue('--experimental-network-inspection')) {
```

Comment:
I think what you need is

```suggestion
  if (!internalBinding('config').hasInspector || !getOptionValue('--experimental-network-inspection')) {
```

instead, because `isEnabled()` is true when the inspector is listening, and this check is done as soon as the process is launched, which means it would not work if the inspector is not activated via e.g. `--inspect`.